### ISSUE TITLE
findOneAndReplace with upsert=true returns an empty document instead of null when the document was created

### ISSUE THREAD

**Issue by ncomet** [bwaldvogel/mongo-java-server#164]
Hi @bwaldvogel,

On version `1.35.0`, or previous, here is what happens, calling with the Java synchronous driver the following code : 

```
collection.findOneAndReplace(
                            Filters.eq("_id", "anId"),
                            updatedDocument,
                            FindOneAndReplaceOptions().upsert(true)
                    )
```

When the document does not previously exists, it does not behave as a regular mongodb database does : mongodb returns `null` when your implementation returns an empty instance of a `Document` (with no attributes). I think it should behave as mongodb returning `null` (even if I would have preferred a null object provided by the driver... but that's another story)

I tried adding the option (which is the default) : 

```
collection.findOneAndReplace(
                            Filters.eq("_id", "anId"),
                            updatedDocument,
                            FindOneAndReplaceOptions().upsert(true).returnDocument(ReturnDocument.BEFORE)
                    )
```

But it also returns an empty document.

There is no problem when adding `ReturnDocument.AFTER` as it returns the Document after the update, but it is not what I would like.

Thanks a lot !

**Comment by Hurryitup** [comment]
I just noticed this happening in `collection.findOneAndUpdate()` as well. As @ncomet, mentioned using `ReturnDocument.AFTER` works fine, but `ReturnDocument.BEFORE` does not.

**Comment by ncomet** [comment]
I forked the project and I reproduce the issue both with findOneAndReplace and findOneAndUpdate, the actual TestContainer test with MongoDB implementation returns `null` when the InMemory impl returns an empty document `{}`, I'm working on it and will try these days to provide a PR.

**Comment by ncomet** [comment]
I fixed the issue in the PR above, first by reproducing with unit tests, then the one line fix, and aligning tests. I think it is ready for review @bwaldvogel.

Thanks !

**Comment by bwaldvogel** [comment]
Thanks for the report and the fix! I’ve just released version `1.36.0` which contains the fix.

**Comment by ncomet** [comment]
Awesome ! Thank you for your reactivity ;) should I close the issue ?
